### PR TITLE
trashItem Model 

### PR DESCRIPTION
**Pull request by sofialovuolo**

### CODE REVIEW

File: server/src/controller/trashController.js [hunk 1]
```diff
@@ -0,0 +1,22 @@
+import validate from "../util/checkValidation.js";
+import prisma from "../../prisma/client.js";
+import { StatusCodes } from "http-status-codes";
+
+export const createTrash = async (req, res) => {
+    if (validate(req, res)) {
+      return res;
+    }
+    
+    const {weight, trashDate, categoryId, userId} = req.body;
```
Comment (ChrisWilhelm) [inline server/src/controller/trashController.js:10]: trashDate is not one of the body arguments listed in the trash.js routes file

**Review by ChrisWilhelm — CHANGES_REQUESTED** [review]
The rest looks good, just one change

File: server/src/controller/trashController.js [hunk 2]
```diff
@@ -0,0 +1,58 @@
...
+    });
+    return res.status(StatusCodes.CREATED).json({ trashItem });
+  };
+
+  export const getTrash = async(req, res) => { 
+    if (validate(req, res)) { //what does this do?
+        return res;
+    }
+    const trash = await prisma.trashItem.findMany({
+        include: {  weight: true, category: true}, //need to talk about what we want to include
```
Comment (jstelnik) [inline server/src/controller/trashController.js:25]: include is for relations so the only things you would include are category and user

File: server/src/controller/trashController.js [hunk 3]
```diff
@@ -0,0 +1,58 @@
...
+    });
+    return res.status(StatusCodes.ACCEPTED).json({ trash });
+  };
+
+  export const editTrash = async(req, res) => {
```
Comment (jstelnik) [inline server/src/controller/trashController.js:30]: you have to get the id from the request so you can call it on line 37

File: server/src/util/trashMiddleware.js [hunk 4]
```diff
@@ -0,0 +1,52 @@
...
+        .status(StatusCodes.CONFLICT)
+        .json({ msg: "ERROR: user does not exist" });
+    } else {
+      next();
+    }
+};
+
+export const isTrashId = async (req, res, next) => { //IDk abt this one
+    const { id } = req.body;
+    const query = await prisma.trashItem.findUnique({ //its .trashItem right?
```
Comment (jstelnik) [inline server/src/util/trashMiddleware.js:40]: yes

File: server/src/routes/trash.js [hunk 5]
```diff
@@ -0,0 +1,64 @@
...
+    body("weight", "PLease include a positive weight deleted!")
+        .notEmpty()
+        .isFloat({gt: 0.0}),
+    body("categoryId", "Please include an integer category Id")
+        .notEmpty()
+        .isInt(),
+    body("userId", "Please include an integer user ID")
+        .notEmpty()
+        .isInt(),
+    //should they be able to edit the date? Like I'd think so but now sure exactly how the date stuff works
```
Comment (jstelnik) [inline server/src/routes/trash.js:46]: yes be able to edit the date so add here

File: server/src/routes/trash.js [hunk 6]
```diff
@@ -0,0 +1,64 @@
...
+        .notEmpty()
+        .isInt(),
+    //should they be able to edit the date? Like I'd think so but now sure exactly how the date stuff works 
+    validator.isCategoryId, 
+    validator.isUserId,
+    controller.createTrash
+);
+
+router.delete(
+    "/:id", //double check this
```
Comment (jstelnik) [inline server/src/routes/trash.js:53]: yes correct

**Review by jstelnik — CHANGES_REQUESTED** [review]

File: server/src/routes/pallet.js [hunk 7]
```diff
@@ -17,7 +17,7 @@ router.post(
     .isInt(),
   body("inputDate", "Please include a date for inputDate").notEmpty().isDate(),
   body("expirationDate", "Expiration Date must be a date").optional().isDate(),
-  body("weight", "Please include a float for entryUserId").notEmpty().isFloat(),
+  body("weight", "Please include an integer for entryUserId").notEmpty().isInt(),
```
Comment (ChrisWilhelm) [inline server/src/routes/pallet.js:20]: This should be a float, the error message should say weight, not entryUserId

File: server/src/routes/trash.js [hunk 8]
```diff
@@ -0,0 +1,64 @@
...
+    validator.isUserId,
+    controller.createTrash
+);
+
+router.delete(
+    "/:id", //double check this 
+    param("id", "Please include an id of the trash to delete")
+        .notEmpty()
+        .isInt(),
+    validator.isTrashId, //in the export.js folder they have this called isTrashIdParam not sure why but I kept this one like this
```
Comment (ChrisWilhelm) [inline server/src/routes/trash.js:57]: The id is coming from the params not the body, you will need to update isTrashId to account for this, or call a new function

**Review by ChrisWilhelm — CHANGES_REQUESTED** [review]

File: server/src/controller/trashController.js (hunk 1 above)
Comment (jstelnik) [reply server/src/controller/trashController.js]: don't we not have to do the date because it is default to now?

**Pull request merged**